test/test.js

```diff
@@ -272,6 +272,14 @@ describe('Page', function() {
     });
   });
 
+  describe('Browser.userAgent', function() {
+    it('should return whether we are in headless', async({browser}) => {
+      const userAgent = await browser.userAgent();
+      expect(userAgent.length).toBeGreaterThan(0);
+      expect(userAgent.includes('Headless')).toBe(headless);
```

Comment:
nit: I'd check that it contains 'webkit' as well (so that the test differs from `browser.getVersion()`)

Reply:
Done
https://github.com/GoogleChrome/puppeteer/pull/1716/commits/28f2626ae79f59060bee1852727ece3ebc4aad12